spec/services/encrypted_doc_storage/doc_writer_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe EncryptedDocStorage::DocWriter do
+  describe '#write' do
+    let(:img_path) { Rails.root.join('app', 'assets', 'images', 'logo.svg')}
+    let(:image) { File.read(img_path) }
+
+    subject do
+      EncryptedDocStorage::DocWriter.new
+    end
+
+    it 'encrypts the document and writes it to storage' do
+      result = subject.write(image:)
+
+      key = Base64.strict_decode64(result.encryption_key)
+      aes_cipher = Encryption::AesCipherV2.new
+
+      written_image = aes_cipher.decrypt(
+        File.read(file_path(result.name)),
+        key,
+      )
+
+      # cleanup
+      File.delete(file_path(result.name))
```

Comment:
another approach to cleanup and writing files to disk would be to make the whole example use a tmpdir, like we do here. We could update the class to take an optional `dir` argument and pass the path in here, the default value would be the `Rails.root.join('tmp')` like have it now

https://github.com/18F/identity-idp/blob/0b303ab9fec96f45c1508e9790aca6330485832a/spec/lib/i18n_flat_yml_backend_spec.rb#L36-L41

Reply:
i considered a couple different approaches to cleanup, but since it was just two specs in different spec files, it seemed fine to just inline it for now. will definitely consider other approaches as we extend it. (unless you feel very strongly about it, then i can update it now!)

Reply:
Not that strongly!